src/test/scala/is/hail/methods/LinearMixedRegressionSuite.scala

```diff
@@ -469,6 +498,24 @@ class LinearMixedRegressionSuite extends SparkSuite {
       vds2.queryGlobal("global.lmmreg.h2")._2.asInstanceOf[Double]))
   }
 
+  def variantAnnotationLMMCompare(vds1: VariantDataset, vds2: VariantDataset) {
+    val vds1Annotations = vds1.variantsAndAnnotations.collect().map{case(variant, annotation) => annotation}
+    val vds2Annotations = vds2.variantsAndAnnotations.collect().map{case(variant, annotation) => annotation}
+
+    val qBeta1 = vds1.queryVA("va.lmmreg.beta")._2
+    val qSg21 = vds1.queryVA("va.lmmreg.sigmaG2")._2
+
+    val qBeta2 = vds2.queryVA("va.lmmreg.beta")._2
+    val qSg22 = vds2.queryVA("va.lmmreg.sigmaG2")._2
+
+    val m = vds1Annotations.length
+
+    (0 until m).foreach { j =>
+      assertDouble(qBeta1(vds1Annotations(j)).asInstanceOf[Double], qBeta2(vds2Annotations(j)).asInstanceOf[Double], 1e-2)
+      assertDouble(qSg21(vds1Annotations(j)).asInstanceOf[Double], qSg22(vds2Annotations(j)).asInstanceOf[Double], 1e-2 )
```

Comment:
It's usually better, but I was seeing very occasional random failures. We should discuss whether this is just a randomization problem where two different random eigendecompositions just are different enough to mess things up, or this is because we aren't doing something dumb.